contracts/upgradeable_contracts/erc20_to_erc20/HomeBridgeErcToErc.sol

```diff
@@ -61,11 +165,39 @@ contract HomeBridgeErcToErc is ERC677Receiver, EternalStorage, BasicBridge, Basi
 
     function onExecuteAffirmation(address _recipient, uint256 _value, bytes32 txHash) internal returns(bool) {
         setTotalExecutedPerDay(getCurrentDay(), totalExecutedPerDay(getCurrentDay()).add(_value));
-        return erc677token().mint(_recipient, _value);
+        uint256 valueToMint = _value;
+        address feeManager = feeManagerContract();
+        if (feeManager != address(0)) {
+            uint256 fee = calculateFee(valueToMint, false, feeManager, FOREIGN_FEE);
+            distributeFeeFromAffirmation(fee, feeManager);
+            emit FeeDistributedFromAffirmation(fee, txHash);
+            valueToMint = valueToMint.sub(fee);
+        }
+        return erc677token().mint(_recipient, valueToMint);
     }
 
     function fireEventOnTokenTransfer(address _from, uint256 _value) internal {
-        emit UserRequestForSignature(_from, _value);
+        uint256 valueToTransfer = _value;
+        address feeManager = feeManagerContract();
+        if (feeManager != address(0)) {
+            uint256 fee = calculateFee(valueToTransfer, false, feeManager, HOME_FEE);
+            valueToTransfer = valueToTransfer.sub(fee);
+        }
+        emit UserRequestForSignature(_from, valueToTransfer);
+    }
+
+    function onSignaturesCollected(bytes _message) internal {
+        address feeManager = feeManagerContract();
+        if (feeManager != address(0)) {
+            address recipient;
+            uint256 amount;
+            bytes32 txHash;
+            address contractAddress;
+            (recipient, amount, txHash, contractAddress) = Message.parseMessage(_message);
+            uint256 fee = calculateFee(amount, true, feeManager, HOME_FEE);
+            distributeFeeFromSignatures(fee, feeManager);
+            emit FeeDistributedFromSignatures(fee, txHash);
```

Comment:
We don't need these event anymore after https://github.com/poanetwork/poa-bridge-contracts/pull/176